modules/solid_properties/test/tests/functional/test.i

```diff
@@ -0,0 +1,115 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 2
+  ny = 2
+[]
+
+[Variables]
+  [u]
+    initial_condition = 1000.0
+  []
+[]
+
+[AuxVariables]
+  [T]
+    initial_condition = 1000.0
+  []
+  [cp]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [k]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [rho]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [dcp_dT]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [dk_dT]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+  [drho_dT]
+    family = MONOMIAL
+    order = CONSTANT
+  []
+[]
+
+[AuxKernels]
+  [cp]
+    type = MaterialRealAux
+    variable = cp
+    property = cp_solid
+  []
+  [k]
+    type = MaterialRealAux
+    variable = k
+    property = k_solid
+  []
+  [rho]
+    type = MaterialRealAux
+    variable = rho
+    property = rho_solid
+  []
+  [dcp_dT]
+    type = MaterialRealAux
+    variable = dcp_dT
+    property = dcp_solid/dT
+  []
+  [dk_dT]
+    type = MaterialRealAux
+    variable = dk_dT
+    property = dk_solid/dT
+  []
+  [drho_dT]
+    type = MaterialRealAux
+    variable = drho_dT
+    property = drho_solid/dT
+  []
+[]
+
+[Materials]
+  [sp_mat]
+    type = ThermalFunctionSolidProperties
```

Comment:
Does the `ThermalFunctionSolidProperties` material support the `output_properties` options? If so, you could list `output_properties = all` or list the names explicitly and AuxKernels and AuxVariables with the same name as the property would be created. This would shorten the input file length.

Reply:
Great idea! I've updated all the tests (I re-golded them because the property names were changed based on the namespace suggestion elsewhere, but I checked that the actual output is exactly the same via exodiff, just with a different name).